vumi/transports/dmark/dmark_ussd.py

```diff
@@ -108,13 +108,48 @@ def session_event_for_transaction(self, transaction_id):
 
     @inlineCallbacks
     def handle_raw_inbound_message(self, request_id, request):
-        values, errors = self.get_field_values(request, self.EXPECTED_FIELDS)
+        try:
+            values, errors = self.get_field_values(request, self.EXPECTED_FIELDS)
+        except UnicodeDecodeError:
+            log.msg('Bad request encoding: %r' % request)
+            request_dict = {
+                'uri': request.uri,
+                'method': request.method,
+                'path': request.path,
+                'content': request.content.read(),
+                'headers': dict(request.requestHeaders.getAllRawHeaders()),
+            }
```

Comment:
Might be worth putting this dictionaryifying into a utility function if we start using it in other transports. Not something for now though.